new-pipeline/src/evo.js

```diff
@@ -412,8 +412,14 @@ function updateAggregates(kind, buckets) {
   }
 
   // Load aggregates from state on first load
-  newAggregates = newAggregates.map(entry => entry[0]);
-  let aggregates = gInitialPageState.aggregates.filter(aggregate => newAggregates.includes(aggregate));
+  let aggregates = gInitialPageState.aggregates
```

Comment:
Could you provide an explanation for what transformation you're performing here? It seems as though you're taking the initial aggregates and... I'm actually a little lost already.

Reply:
I think I get it now. We want a list of newAggregates' keys where either the key or the value of the newAggregate is present in gInitialPageState.aggregates. Is that right? I think we can make this a bit more obvious if we write it something like:
```js
let aggregates = [];
for (let [key, value] of newAggregates) {
  if (gInitialPageState.aggregates.includes(key) || gInitialPageState.aggregates.includes(value)) {
    aggregates.push(key);
  }
}
```